Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -371,22 +371,53 @@ TicketReader::TicketReader(std::vector<u8> bytes) : SignedBlobReader(std::move(b
 
 bool TicketReader::IsValid() const
 {
-  return IsSignatureValid() && !m_bytes.empty() && m_bytes.size() % sizeof(Ticket) == 0;
+  if (!IsSignatureValid() || m_bytes.empty())
+    return false;
+
+  if (IsV1Ticket())
+  {
+    // V1 Ticket size is normal ticket size + the v1 ticket size specified in the v1 header.
+    const u32 ticket_size =
+        Common::swap32(m_bytes.data() + sizeof(Ticket) + offsetof(V1TicketHeader, v1_ticket_size)) +
+        sizeof(Ticket);
+    return m_bytes.size() == ticket_size;
+  }
+
+  return m_bytes.size() % sizeof(Ticket) == 0;
+}
+
+bool TicketReader::IsV1Ticket() const
+{
+  // Version can only be 0 or 1.
+  return m_bytes[offsetof(Ticket, version)] == 1;
 }
 
 size_t TicketReader::GetNumberOfTickets() const
 {
+  if (IsV1Ticket())
+    return 1;
+
   return m_bytes.size() / sizeof(Ticket);
 }
 
+u32 TicketReader::GetTicketSize() const
+{
+  if (IsV1Ticket())
```

Comment:
Normally we brace if statements where the body travels more than one line, to clearly delineate the scope of the if statement and make it easier to see